### ISSUE TITLE
How to obtain class-wise metrics for multi-class segmentation

### ISSUE THREAD

**Issue by gizemtanriver** [qubvel/segmentation_models.pytorch#327]
Hi @qubvel, I am working on a dataset with 3 classes + 1 background class. I obtained the IoU and F1 scores on the test set, but I also want to know the results for each class. How can I obtain this? 

For instance, when I use ignore_channels to get results for class label 2 (which corresponds to the 2nd channel in the produced mask), I get quite low results for all classes except background so not sure if this is the right way.

smp.utils.metrics.IoU(threshold=0.5, ignore_channels=[0,1,3])

Many thanks.

**Comment by johschmidt42** [comment]
Hi @gizemtanriver, you can use the metrics API from different libraries to do that, e.g. [sklearn](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics), [pytorch-lightning](https://pytorch-lightning.readthedocs.io/en/latest/metrics.html#functional-metrics-classification), [pytorch-ignite](https://pytorch.org/ignite/metrics.html) and many more. There's often an argument like `average` or `reduction` that you should set to `None` or `'none'`. 

Here's an example for PL and sklearn:

```
import torch

batch_size = 1
size = (128, 128)
num_classes = 4

logits = torch.randn(size=(batch_size, num_classes) + size)  # network's output [N, C, H, W]
output_softmax = torch.softmax(logits, dim=1)  # softmax on logits
output_argmax = torch.argmax(output_softmax, dim=1)

target = torch.randint(low=0, high=num_classes, size=(batch_size,) + size)

from sklearn.metrics import f1_score

f1_sklearn = f1_score(target.flatten(), output_argmax.flatten(), average=None)
# array([0.24566121, 0.26164661, 0.24077285, 0.2507698 ])

from pytorch_lightning.metrics.functional import f1
f1_pl = f1(preds=output_softmax, target=target, num_classes=num_classes, average='none')
# tensor([0.2457, 0.2616, 0.2408, 0.2508])
```

**Comment by ramchandracheke** [comment]
Hi, 

I found solution as, Once the instance of train_epoch is created then we can change name as per below mentioned code.

```python

metrics = [smp.utils.metrics.IoU(threshold=0.5, ignore_channels=[1,2,3]),
              smp.utils.metrics.IoU(threshold=0.5, ignore_channels=[0,2,3]),
              smp.utils.metrics.IoU(threshold=0.5, ignore_channels=[0,1,3])
]

train_epoch = smp.utils.train.TrainEpoch(
    model, 
    loss=loss, 
    metrics=metrics, 
    optimizer=optimizer,
    device=DEVICE,
    verbose=True,
)

train_epoch.metrics[0].__name__="IoU_Class0"
train_epoch.metrics[1].__name__="IoU_Class1"
train_epoch.metrics[2].__name__="IoU_Class2"
```

This will be helpful to create IoU score at class level.
